content/blog/2023/2023-05-18-operators-in-typescript.md

```diff
@@ -0,0 +1,177 @@
+
+{{% github "https://github.com/Irtaza2009/code-examples/tree/operators-in-typescript/nodejs/operators-in-typescript" %}}
+
+# What operators are in Typescript? How are they used?
+In Typescript, **operators are symbols used to perform operations on variables or values.** They can be classified into several categories based on their functions.
+
+## Concatenation Operators
+Concatenation operators in TypeScript are used to combine strings and values together. The most common concatenation operator is the plus sign (+). When used with strings, the plus sign combines them into a single string. When used with a string and a value, the plus sign concatenates the value to the end of the string.
+
+For example, let's say we have two strings, "Hello" and "World". We can use the concatenation operator to combine them into a single string:
+
+```ts
+let greeting = "Hello" + "World";
+
+console.log(greeting);
+```
+Output:
+~~~ts
+HelloWorld
+~~~
+
+We can also use the concatenation operator to combine a string and a value:
+
+~~~ts
+let age = 30;
+let message = "I am " + age + " years old.";
+console.log(message); 
+~~~
+Output:
+~~~ts
+I am 30 years old.
+~~~
+
+In this example, the concatenation operator combines the string "I am " with the value of the age variable (30) and the string " years old." to create the final message.
+
+Concatenation operators are useful in situations where we need to build dynamic strings based on values or user input. By using concatenation operators, we can create custom messages or outputs that are tailored to our specific needs.
+
+## Arithmetic Operators
+Arithmetic operators allow us to perform mathematical operations such as addition, subtraction, multiplication, division etc on numerical values (constants and variables). Let’s take a look at them:
+
+~~~ts
+let x = 5;
+let y = 10;
+
+console.log(x + y); // Output: 15
+console.log(x - y); // Output: -5
+console.log(x * y); // Output: 50
+console.log(x / y); // Output: 0.5
+console.log(y % x); // Output: 0
+
+let z = 3;
+z++; 
+console.log(z);   // Output: 4
+
+let a = 10;
+a--;
+console.log(a);   // Output: 9
+~~~
+
+**Addition (`+`):** adds two or more values
+
+**Subtraction (`-`):** subtracts two or more values
+
+**Multiplication (`*`):** multiplies two or more values
+
+**Division (`/`):** divides two or more values
```

Comment:
What's the precision of the result? What happens if I divide two integers compared to dividing two floats compared to diving an integer by a float or a float by an integer?